core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java

```diff
@@ -72,7 +72,7 @@ public void resetMetaMasterConfigClient(MasterClientContext context) {
   }
 
   @Override
-  public synchronized void heartbeat() {
+  public void heartbeat() {
```

Comment:
`mClient` is used in this method.. so what if another thread closes it before an operation on `mClient`?

Reply:
mClient is also threadsafe by having its own synchronized methods.

Reply:
mClient being synchronized does not help right? 
`mClient.close()` runs first, and then `mClient.someRPCCall()` or `mClient.connect()`  would fail and throw a `FailedPreconditionException` which is not handled gracefully..